### ISSUE TITLE
shortcode-content-image action/hook suggestion

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/easy-digital-downloads#5957]
I am building a custom EDD theme and had to create a custom template for the `shortcode-content-image.php` file that included a hook.

Since I wanted to position the review stars (and wish list buttons) over the image, I added the following:

```
<?php if ( function_exists( 'has_post_thumbnail' ) && has_post_thumbnail( get_the_ID() ) ) : ?>
	<div class="edd_download_image">
		<a href="<?php the_permalink(); ?>">
			<?php echo get_the_post_thumbnail( get_the_ID(), 'thumbnail' ); ?>
		</a>
		<div class="edd_download_image_overlay">
			<?php do_action( 'edd_download_image_overlay' ); ?>
		</div>
	</div>
<?php endif; ?>
```
For obvious reasons the divs can be excluded if EDD chose to add a hook here.

**Comment by cklosowski** [comment]
@SDavisMedia @spencerfinnell Both of you have written themes (Vendd and Marketify respectively), that use a custom template file for the content image.

Does adding a hook here seem like something that's 'standard' from a theming perspective or is the answer typically that it should be on the theme to make their own template file (setting a precedence would be my concern).

The hook seems like it would end up getting used in plugins as well, which might not be ideal for all themes that could be using EDD like Vendd or Marketify. My main concern is adding in a hook here could alter a theme pretty unintentionally if a plugin author chooses to hook into it. While Vendd and Marketify wouldn't necessarily have this hook included because it's something they'd have to back-port...

**Comment by SDavisMedia** [comment]
I don't think this is a bad request, but it is definitely incomplete. There's no reason to add just one hook for something like this. If we'd add anything, it would be strategically placed hooks throughout the HTML structure and in all of the overridable templates (some already have a few hooks).

Aren't we changing this template system soon? If so, the last thing we need is several hooks being used by theme and plugin devs. Having the templates overridden by several users is already "bad" enough. 

If the system is changing soon, I think changes should wait for that.

**Comment by davidsherlock** [comment]
I see what you are saying @SDavisMedia, certainly more than one hook could be added but for the time being I will simply handle this via a modified template in my EDD theme.

**Comment by SDavisMedia** [comment]
:+1: We'll definitely keep hook placement in mind if the system is changing. If not, it might not be a bad idea to sweep through and add some hooks to all of the files.
